src/schematools/types.py

```diff
@@ -754,18 +793,18 @@ def provenance(self) -> Optional[str]:
         return self.get("provenance")
 
     @property
-    def items(self) -> Dict[str, Any]:
+    def field_items(self) -> Optional[Dict[str, Any]]:
         """Return the item definition for an array type."""
         return self.get("items", {}) if self.is_array else None
 
     def get_dimension_fieldnames_for_relation(
-        self, relation: str, nm_relation: str
+        self, relation: Optional[str], nm_relation: Optional[str]
     ) -> Dict[str, List[str]]:
         """Gets the dimension fieldnames."""
         if relation is None and nm_relation is None:
             return {}
 
-        dataset_id, table_id = (relation or nm_relation).split(":")
+        dataset_id, table_id = str(relation or nm_relation).split(":")
```

Comment:
Hmm, so I presume MyPy can't determine, from the above if statement, that this or-expression evaluates to a `str`? Weird.

Reply:
Using cast now.